deploy/kustomize/deploy_kustomize.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+
+set -x
+
+# Setup Environment Variables
+HCO_VERSION="${HCO_VERSION:-1.1.0}"
+HCO_CHANNEL="${HCO_CHANNEL:-1.1.0}"
+MARKETPLACE_MODE="${MARKETPLACE_MODE:-true}"
+HCO_REGISTRY_IMAGE="${HCO_REGISTRY_IMAGE:-quay.io/kubevirt/hco-container-registry:latest}"
+PRIVATE_REPO="${PRIVATE_REPO:-false}"
+QUAY_USERNAME="${QUAY_USERNAME:-}"
+QUAY_PASSWORD="${QUAY_PASSWORD:-}"
+CONTENT_ONLY="${CONTENT_ONLY:-false}"
+KVM_EMULATION="${KVM_EMULATION:-false}"
+OC_TOOL="${OC_TOOL:-oc}"
+
+#####################
+
+main() {
+  SCRIPT_DIR="$(dirname "$0")"
+  TARGET_NAMESPACE=$(grep name: $SCRIPT_DIR/namespace.yaml | awk '{print $2}')
+
+  TMPDIR=$(mktemp -d)
+  cp -r $SCRIPT_DIR/* $TMPDIR
+
+  if [ "$PRIVATE_REPO" = 'true' ]; then
+    get_quay_token
+    oc create secret generic quay-registry-kubevirt-hyperconverged --from-literal=token="$QUAY_TOKEN" -n openshift-marketplace
+
+    cat <<EOF >$TMPDIR/kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+bases:
+  - private_repo
+EOF
+    oc apply -k $TMPDIR
+
+  else # not private repo
+    if [ "$MARKETPLACE_MODE" = 'true' ]; then
+      cat <<EOF >$TMPDIR/kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+bases:
+  - marketplace
+EOF
+      oc apply -k $TMPDIR
+    else
+      cat <<EOF >$TMPDIR/kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+bases:
+  - image_registry
+EOF
+      oc apply -k $TMPDIR
+    fi
+  fi
+
+  if [ "$CONTENT_ONLY" = 'true' ]; then
+    echo INFO: Content is ready for deployment in OLM.
+    exit 0
+  fi
+
+  # KVM_EMULATION setting is active only when a deployment is done.
+  if [ "$KVM_EMULATION" = 'true' ]; then
+    cat <<EOF >$TMPDIR/kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: ${TARGET_NAMESPACE}
+bases:
+  - kvm_emulation
+resources:
+  - namespace.yaml
+EOF
+    exit
+    retry_loop $TMPDIR
+  else
+    # In case KVM_EMULATION is not set.
+    cat <<EOF >$TMPDIR/kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: ${TARGET_NAMESPACE}
+bases:
+  - base
+resources:
+  - namespace.yaml
+EOF
+    retry_loop $TMPDIR
+  fi
+}
+
+get_quay_token() {
+    token=$(curl -sH "Content-Type: application/json" -XPOST https://quay.io/cnr/api/v1/users/login -d '
+  {
+      "user": {
+          "username": "'"${QUAY_USERNAME}"'",
+          "password": "'"${QUAY_PASSWORD}"'"
+      }
+  }' | jq -r '.token')
+
+  if [ "$token" == "null" ]; then
+    echo [ERROR] Got invalid Token from Quay. Please check your credentials in QUAY_USERNAME and QUAY_PASSWORD.
+    exit 1
+  else
+    QUAY_TOKEN=\"$token\";
+  fi
+}
+
+# Deploy HCO and OLM Resources with retries
+retry_loop() {
```

Comment:
@dankenigsberg just FYI - this is a retryloop we need to mitigae the not-auto-create-cr

Reply:
Yes, I'm aware of it. If we auto-create, users with default requirements could avoid this retry_loop. But we cannot avoid in code if we want to support user kustomization of the HCO.cr. We currently have nothing there, but sooner or later, we would.